tests/unit/transaction_queue.cpp

```diff
@@ -0,0 +1,106 @@
+
+/*
+Tests rely on the fact that interpreters are sequentially added to runninng_interpreters to get transaction_id of its
+corresponding interpreter/.
+*/
+class TransactionQueueSimpleTest : public ::testing::Test {
+ protected:
+  memgraph::storage::Storage db_;
+  std::filesystem::path data_directory{std::filesystem::temp_directory_path() / "MG_tests_unit_transaction_queue_intr"};
+  memgraph::query::InterpreterContext interpreter_context{&db_, {}, data_directory};
+  InterpreterFaker running_interpreter{&interpreter_context}, main_interpreter{&interpreter_context};
+};
+
+// This test relies on explicit transations + explicit thread killing for testing if show transactions work as expected
+TEST_F(TransactionQueueSimpleTest, ShowTransactions) {
+  std::jthread running_thread = std::jthread(
+      [this](int thread_index) {
+        running_interpreter.Interpret("BEGIN");
+        for (int i = 0; i < 5; i++) {
+          running_interpreter.Interpret("CREATE (:Person {prop: " + std::to_string(thread_index) + "})");
+          std::this_thread::sleep_for(std::chrono::milliseconds(200));
+        }
+        running_interpreter.Interpret("ROLLBACK");
+      },
+      0);
+
+  {
+    main_interpreter.Interpret("CREATE (:Person {prop: 1})");
+    auto show_stream = main_interpreter.Interpret("SHOW TRANSACTIONS");
+    ASSERT_EQ(show_stream.GetResults().size(), 2U);
+    // admin executing the transaction
+    EXPECT_EQ(show_stream.GetResults()[0][0].ValueString(), "");
+    ASSERT_TRUE(show_stream.GetResults()[0][1].IsString());
+    EXPECT_EQ(show_stream.GetResults()[0][2].ValueList().at(0).ValueString(), "SHOW TRANSACTIONS");
+    // Also admin executing
+    EXPECT_EQ(show_stream.GetResults()[1][0].ValueString(), "");
+    ASSERT_TRUE(show_stream.GetResults()[1][1].IsString());
+    EXPECT_EQ(show_stream.GetResults()[1][2].ValueList().at(0).ValueString(), "EXPLICIT TRANSACTION");
+    // test the state of the database
+    auto results_stream = main_interpreter.Interpret("MATCH (n) RETURN n");
+    ASSERT_EQ(results_stream.GetResults().size(), 1U);
+    main_interpreter.Interpret("MATCH (n) DETACH DELETE n");
+  }
+}
+
+TEST_F(TransactionQueueSimpleTest, TerminateTransaction) {
+  std::jthread running_thread = std::jthread(
+      [this](std::stop_token st, int thread_index) {
+        running_interpreter.Interpret("BEGIN");
+        while (!st.stop_requested()) {
+          running_interpreter.Interpret("CREATE (:Person {prop: " + std::to_string(thread_index) + "})");
+          std::this_thread::sleep_for(std::chrono::milliseconds(200));
+        }
+      },
+      0);
+
+  {
+    main_interpreter.Interpret("CREATE (:Person {prop: 1})");
+    auto show_stream = main_interpreter.Interpret("SHOW TRANSACTIONS");
+    ASSERT_EQ(show_stream.GetResults().size(), 2U);
+    // superadmin executing the transaction
+    EXPECT_EQ(show_stream.GetResults()[0][0].ValueString(), "");
+    ASSERT_TRUE(show_stream.GetResults()[0][1].IsString());
```

Comment:
Is there a way to check this ID also?

Reply:
That would test storage, I don't think this this has nothing to do with transaction_queue and if the test passes that means that the correct transaction with the correct transaction_id has been killed

Reply:
agree